android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -830,4 +898,276 @@ constructor(
+                        syncParamSource.compListRequestParamForItems.add(ResTypeId(ResourceType.fromCode(resourceType), it))
+                      }
+                    }
+                  }
+
+                  Timber.d("configSync compListRequestQue size ${syncParamSource.compListRequestQue.size}")
+                  fetchCompositionListRequest(coroutineScope)
+
+                } else {
+
+                  entry.value.forEach {
+                    val compositionManifestRequestParamPairs = mutableListOf<Pair<ResourceType, Map<String, String>>>()
+                    Timber.d("configSync composition Manifest entry.key ${entry.key} and id= ${it.focus.extractId()}")
+                    compositionManifestRequestParamPairs.add(
+                      Pair(
+                        ResourceType.fromCode(entry.key),
+                        mapOf(Composition.SP_RES_ID to it.focus.extractId()),
+                      ),
+                    )
+                    syncParamSource.compManifestRequestQue.push( mapOf(*compositionManifestRequestParamPairs.toTypedArray()))
+                  }
+
+                  Timber.d("configSync composition Manifest RequestQue size - ${syncParamSource.compManifestRequestQue.size}")
+                  fetchCompositionManifestRequest(coroutineScope)
+
+                }
+              } else {
+                Timber.d("configSync process compositionContent if Not LIST block")
+                val chunkedResourceIdList = entry.value.chunked(MANIFEST_PROCESSOR_BATCH_SIZE)
+                chunkedResourceIdList.forEach { parentIt ->
+                  val resourceIds: List<String> =
+                    parentIt.map { sectionComponent -> sectionComponent.focus.extractId() }
+                  val resourceType = entry.key
+
+                  resourceIds.forEach {
+                    Timber.d("configSync composition Config adding resourceType - ${resourceType} and id= ${it}")
+                    val compConfigParamPairsReq = mutableListOf<Pair<ResourceType, Map<String, String>>>()
+                    compConfigParamPairsReq.add(
+                      Pair(
+                        ResourceType.fromCode(resourceType),
+                        mapOf(Composition.SP_RES_ID to it),
+                      ),
+                    )
+                    syncParamSource.compConfigRequestQue.push(mapOf(*compConfigParamPairsReq.toTypedArray()))
+                  }
+
+                  Timber.d("configSync composition Config RequestQue size - ${syncParamSource.compConfigRequestQue.size}")
+
+                  fetchCompositionConfigRequest(coroutineScope)
+                }
+              }
+            }
+            saveSyncSharedPreferences(patientRelatedResourceTypes.toList())
+        }
+      }
+    }
+  }
+
+  private suspend fun fetchCompositionManifestRequest(coroutineScope: CoroutineScope) {
+
+    val compositionManifestFlow4 = Sync.oneTimeSync<CompositionManifestSyncWorker>(context)
```

Comment:
we can remove the No. 4 suffix here since it is the only flow in this methods scope

Reply:
updated